src/Ship.js

```diff
@@ -0,0 +1,39 @@
+function Ship(itinerary) {
+   this.itinerary = itinerary;
+   this.currentPort = itinerary.ports[0];
+   this.previousPort = null;
+   this.currentPort.addShip(this);
+
+};
+
+Ship.prototype.setSail = function() {
+   const itinerary = this.itinerary;
+   const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+   if(currentPortIndex === (itinerary.ports.length -1)) {
+    throw new Error('End of itinerary reached');
+   }
+   this.previousPort = this.currentPort;
+   this.currentPort = null;
+   this.previousPort.removeShip(this);
+    
+};
```

Comment:
A few things here, mostly formatting, but also you dont need to declare the itinerary variable you are here when you can just reference `this.itinerary`:
```
Ship.prototype.setSail = function () {
    const currentPortIndex = this.itinerary.ports.indexOf(this.currentPort);

    if (currentPortIndex === this.itinerary.ports.length - 1) {
        throw new Error("End of itinerary reached");
    }

    this.previousPort = this.currentPort;
    this.currentPort = null;
    this.previousPort.removeShip(this);
};
```